src/System.Buffers/src/System/Buffers/BufferPool.cs

```diff
@@ -41,7 +36,8 @@ public T[] RentBuffer(int size, bool clearBuffer = false)
             }
             else
             {
-                buffer = _buckets[SelectBucketIndex(size)].Rent();
+                int index = SelectBucketIndex(size);
+                buffer = LazyInitializer.EnsureInitialized(ref _buckets[index], () => new BufferBucket<T>(GetMaxSizeForBucket(index), NUMBER_OF_BUFFERS_IN_BUCKET)).Rent();
```

Comment:
> Would something like that allocate? (if delegate allocates then we can't avoid it but it feels like this shouldn't be)

Yes, there will be a delegate allocation, in order to create a delegate from the instance method you're passing in as a method group.